src/util/geometry.c

```diff
@@ -76,7 +98,15 @@ bool lineseg_ellipse_intersect(LineSegment seg, Ellipse e) {
 	seg.b = creal(seg.b) + I * ratio * cimag(seg.b);
 
 	Circle c = { .radius = creal(e.axes) / 2 };
-	return lineseg_circle_intersect(seg, c) >= 0;
+	return lineseg_circle_intersect_fallback(seg, c) >= 0;
+}
+
+double lineseg_circle_intersect(LineSegment seg, Circle c) {
+	Ellipse e = { .origin = c.origin, .axes = c.radius + I*c.radius };
```

Comment:
Shouldn't the radius be multiplied by 2 here, since the axes aren't half-axes?

Reply:
Yes, fixed.